### PR TITLE
Add mirror method for getPersistenceNamingEncoding

### PR DESCRIPTION
**Pull request by Renkai**
This method will be used for where we need to get Schema info from the persistent name in some plugins.

### CODE REVIEW

File: pulsar-common/src/main/java/org/apache/pulsar/common/naming/TopicName.java [hunk 1]
```diff
@@ -320,6 +322,33 @@ public String getPersistenceNamingEncoding() {
         }
     }
 
+    public static TopicName fromPersistenceNamingEncoding(String name) throws Exception {
```
Comment (zymap) [inline pulsar-common/src/main/java/org/apache/pulsar/common/naming/TopicName.java:325]: I'm not sure what is this used for? Looks like we can parse it in the plugin and avoid introducing it into Pulsar?

File: pulsar-common/src/test/java/org/apache/pulsar/common/naming/TopicNameTest.java [hunk 2]
```diff
@@ -162,6 +161,9 @@ public void topic() {
         assertEquals(TopicName.get("persistent://tenant/cluster/namespace/topic")
                 .getPersistenceNamingEncoding(), "tenant/cluster/namespace/persistent/topic");
 
+        assertEquals(TopicName.fromPersistenceNamingEncoding("tenant/cluster/namespace/persistent/topic"),
```
Comment (zymap) [inline pulsar-common/src/test/java/org/apache/pulsar/common/naming/TopicNameTest.java:164]: You may need to add a case for using the `getPersistenceNamingEncoding` returned value to parse to ensure the encoded value can be parsed successfully.

File: pulsar-common/src/main/java/org/apache/pulsar/common/naming/TopicName.java (hunk 1 above)
Comment (Renkai) [reply pulsar-common/src/main/java/org/apache/pulsar/common/naming/TopicName.java:325]: It's not for "one plugin", but maight be used for various plugins which may need to parse managed ledger name

File: pulsar-common/src/main/java/org/apache/pulsar/common/naming/TopicName.java [hunk 3]
```diff
@@ -320,6 +322,33 @@ public String getPersistenceNamingEncoding() {
...
+            return null;
+        }
+        final String[] arr = name.split("/");
+
+        if (arr.length == 4) {
+            String tenant = arr[0];
+            String namespacePortion = arr[1];
+            String domain = arr[2];
+            String encodedLocalName = arr[3];
+            final String decodedName = Codec.decode(encodedLocalName);
```
Comment (codelipenghui) [inline pulsar-common/src/main/java/org/apache/pulsar/common/naming/TopicName.java:336]: Looks this case does not cover by any test? Could you please also add a test for this case?

File: pulsar-common/src/main/java/org/apache/pulsar/common/naming/TopicName.java [hunk 4]
```diff
@@ -320,6 +322,33 @@ public String getPersistenceNamingEncoding() {
...
+            String tenant = arr[0];
+            String cluster = arr[1];
+            String namespacePortion = arr[2];
+            String domain = arr[3];
+            String encodedLocalName = arr[4];
+            final String decodedName = Codec.decode(encodedLocalName);
+            return TopicName.get(domain, tenant, cluster, namespacePortion, decodedName);
+        } else {
+            log.error("arr.length = {}, arr = {}", arr.length, Stream.of(arr).collect(Collectors.toList()));
+            throw new Exception("not valid name: " + name);
```
Comment (315157973) [inline pulsar-common/src/main/java/org/apache/pulsar/common/naming/TopicName.java:348]: No unit test coverage for this branch

**Comment by codelipenghui** [comment]
The pr had no activity for 30 days, mark with Stale label.

**Comment by tisonkun** [comment]
Closed as stale. Please rebase and resubmit the patch if it's still relevant.

**Pull request closed without merging**